racket/collects/racket/syntax.rkt

```diff
@@ -205,6 +205,7 @@
 
 (define-syntax (with-syntax* stx)
   (syntax-case stx ()
+    [(_ () body ...) #'(begin body ...)]
```

Comment:
Should this be let instead of begin?

Reply:
According to the docs, yes.

I though about this, but after looking at the implementation of `with-syntax` (without the star), I think that `with-syntax` (without the star) use `begin` instead of `let()`.

Perhaps we should change the implementation of `with-syntax` (without the star), or perhaps I made a mistake reading it.

Reply:
After thinking it for another while, let's change both `with-syntax` and `with-syntax*`.